pyproject.toml

```diff
@@ -124,6 +124,17 @@ tracker = "https://github.com/scipy/scipy/issues"
 [tool.doit]
 dodoFile = "dev.py"
 
+[tool.meson-python]
```

Comment:
We shouldn't need `tool.meson-python` changes in this PR, and definitely don't want to vendor Meson. Can you drop these changes? Everything else should just work I believe.

Reply:
Sure.

Reply:
A ping on this - testing this branch is a little easier when this comment is addressed. The git submodule is still present

Reply:
Deleted this `tool.meson-python`. Also deleted `vendored-meson/meson` completely from here. No problems should be there now. I am resolving this thread for the time being. Let me know if you still face problems.